SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/SteamUnifiedMessages.cs

```diff
@@ -158,101 +71,65 @@ public AsyncJob<ServiceMethodResponse> SendMessage<TRequest>( string name, TRequ
-            {
-                case EMsg.ServiceMethodResponse:
-                    HandleServiceMethodResponse( packetMsg );
-                    break;
-
-                case EMsg.ServiceMethod:
-                    HandleServiceMethod( packetMsg );
-                    break;
-            }
-        }
-
-
-        #region ClientMsg Handlers
-        void HandleServiceMethodResponse( IPacketMsg packetMsg )
-        {
-            if ( packetMsg is not PacketClientMsgProtobuf packetMsgProto )
-            {
-                throw new InvalidDataException( "Packet message is expected to be protobuf." );
-            }
-
-            var callback = new ServiceMethodResponse( packetMsgProto );
-            Client.PostCallback( callback );
-        }
-
-        void HandleServiceMethod( IPacketMsg packetMsg )
-        {
-            if ( packetMsg is not PacketClientMsgProtobuf packetMsgProto )
-            {
-                throw new InvalidDataException( "Packet message is expected to be protobuf." );
-            }
+            if (packetMsg is not PacketClientMsgProtobuf packetMsgProto)
+                return;
 
-            var jobNameStr = packetMsgProto.Header.Proto.target_job_name;
-            if ( string.IsNullOrEmpty( jobNameStr ) )
-            {
+            var jobName = packetMsgProto.Header.Proto.target_job_name.AsSpan();
+            if (jobName.IsEmpty)
                 return;
-            }
 
             // format: Service.Method#Version
-            var jobName = jobNameStr.AsSpan();
             var dot = jobName.IndexOf( '.' );
             var hash = jobName.LastIndexOf( '#' );
             if ( dot < 0 || hash < 0 )
-            {
                 return;
-            }
 
             var serviceName = jobName[ ..dot ].ToString();
+            if (!_handlers.TryGetValue( serviceName, out var handler ) )
+                return;
+
             var methodName = jobName[ ( dot + 1 )..hash ].ToString();
 
-            var serviceInterfaceName = "SteamKit2.Internal.I" + serviceName;
-            var serviceInterfaceType = Type.GetType( serviceInterfaceName );
-            if ( serviceInterfaceType != null )
-            {
-                var method = serviceInterfaceType.GetMethod( methodName );
-                if ( method != null )
-                {
-                    var argumentType = method.GetParameters().Single().ParameterType;
+            handler.HandleMsg( methodName, packetMsg );
+        }
 
-                    var callback = new ServiceMethodNotification( argumentType, packetMsg );
-                    Client.PostCallback( callback );
-                }
+        internal void HandleServiceMsg<TService>( IPacketMsg packetMsg )
+            where TService : IExtensible, new()
+        {
+            var callback = new ServiceMsg<TService>( (packetMsg as PacketClientMsgProtobuf)! );
+            Client.PostCallback( callback );
+        }
+
+        public abstract class UnifiedService : IDisposable
```

Comment:
need xmldocs. I think these HandleMsg and ServiceName likely need to be public for consumers to be able to implement their own service handlers (e.g. try generating a service proto outside of SteamKit project)